File: DS -py/DetectCycleLinkedList.py
class Node:
    def __init__(self,data):
        self.next=None;
        self.data=data

def LinkedListValues():
    root=Node(1)
    t=root
    for i in range(5,10):
        tmp=Node(i)
        t.next=tmp
        t=t.next
    #Connecting tail to head
    t.next=root
    return root

def CheckCycle(Node):
    slow=fast=Node
    while slow!=None and fast!=None:
        if fast.next==slow:
            return True
        slow=slow.next
        fast=fast.next.next
    return False

root=LinkedListValues()
if CheckCycle(root)==True:
    print("Cycle Detected !")
else:
    print("No Cycle")

        

File: DS/DetectCycleLinkedList.cpp
#include<bits/stdc++.h>
using namespace std;
struct Node{
Node *next;
int data;
Node(int x){
    data=x;
    next=NULL;
}
};

Node* LinkedListValues(){
//Creating a linked list with a cycle
Node *head=new Node(1);
Node *t=head;
for(int i=5;i<10;i++){
Node *tmp=new Node(i);
t->next=tmp;
t=t->next;
}
//connecting tail to head
t->next=head;
return head;
}

bool CheckCycle(Node *head){
Node *slow=head;
Node *fast=head;
while(slow && fast){
    if(fast->next==slow) return true;
    slow=slow->next;
    fast=fast->next->next;
}
return false;
}

int main(){
Node *root=LinkedListValues();
if(CheckCycle(root)==true) cout<<"Cycle dectected !";
else cout<<"No Cycle";
return 0;
}